### PR TITLE
Fix AnsiCastShim for 330db [databricks]

### PR DESCRIPTION
**Pull request by andygrove**
Closes https://github.com/NVIDIA/spark-rapids/issues/8164

Databricks 11.3 apparently does not have the `AnsiCast` expression from Spark 3.3.x

Status:

- [x] Confirm that this fixes the compilation issue
- [ ] Confirm that tests pass (in progress, but we should probably go ahead and merge this anyway)

### CODE REVIEW

File: sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/AnsiCastShim.scala
Comment (jlowe) [inline sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/AnsiCastShim.scala:1]: This file looks the same as spark340 version?

Comment (andygrove) [reply sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/AnsiCastShim.scala:1]: It is. Would it be preferable just to add the 330db tag in the 340 shims instead?

Comment (jlowe) [reply sql-plugin/src/main/spark330db/scala/com/nvidia/spark/rapids/shims/AnsiCastShim.scala:1]: Yes. IIRC shimplify has some logic to complain if shim files are identical and should have been shared.

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Comment by andygrove** [comment]
I just ran into this verifying the build on databricks

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.1.0:run (shimplify-shim-sources) on project rapids-4-spark-sql_2.12: An Ant BuildException has occured: Traceback (most recent call last):
[ERROR]   File "<script>", line 571, in <module>
[ERROR]   File "<script>", line 351, in task_impl
[ERROR]   File "<script>", line 365, in __generate_symlinks
[ERROR]   File "<script>", line 394, in __traverse_source_tree_of_all_shims
[ERROR]   File "<script>", line 365, in <lambda>
[ERROR]   File "<script>", line 407, in __generate_symlink_to_file
[ERROR] AssertionError: Unexpected: spark340/scala/com/nvidia/spark/rapids/shims/AnsiCastShim.scala is not prefixed by spark330db/
```

**Comment by jlowe** [comment]
> I just ran into this verifying the build on databricks

Ah, right, shimplify wants the file to be in the shim directory of the lexicographically lowest version, so needs to move from spark340/scala to spark330db/scala

**Comment by gerashegalov** [comment]
> I just ran into this verifying the build on databricks
> 
> ```
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-antrun-plugin:3.1.0:run (shimplify-shim-sources) on project rapids-4-spark-sql_2.12: An Ant BuildException has occured: Traceback (most recent call last):
> [ERROR]   File "<script>", line 571, in <module>
> [ERROR]   File "<script>", line 351, in task_impl
> [ERROR]   File "<script>", line 365, in __generate_symlinks
> [ERROR]   File "<script>", line 394, in __traverse_source_tree_of_all_shims
> [ERROR]   File "<script>", line 365, in <lambda>
> [ERROR]   File "<script>", line 407, in __generate_symlink_to_file
> [ERROR] AssertionError: Unexpected: spark340/scala/com/nvidia/spark/rapids/shims/AnsiCastShim.scala is not prefixed by spark330db/
> ```

shimplify requires the shim class reside with the owner shim https://github.com/NVIDIA/spark-rapids/blob/branch-23.06/docs/dev/shimplify.md. It's currently defined as the smallest buildver in the lexicographical order. in this case 330db < 340. so the file should be moved to spark330db

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Review by gerashegalov — APPROVED** [review]
LGTM

**Pull request merged**